src/node.js

```diff
@@ -206,6 +207,13 @@ class Node {
             this.api.apiHost = addr.address
             this.api.apiPort = addr.port
 
+            if (gwmatch) {
+              this.gatewayAddr = gwmatch[2]
+              const addr = multiaddr(this.gatewayAddr).nodeAddress()
```

Comment:
Changed the code now to expose `apiAddr` and `gatewayAddr` in the `Node` and to keep api compatibility, keep the separate variables for host and port in the `ipfs` instance. I'm not a huge fan of the monkey-patching solution here as it changed the public API of js-ipfs-api, but that's how it's always been, so I don't want to break that now.

@dignifiedquire take a look at let me know if you think this is good to go.